### PR TITLE
adding query syntax

### PR DESCRIPTION
**Pull request by btel**
Needs some extra testing!

Example package spec:

```
"my-package>=0.1.2,<0.2,other-package==1.1"
```

which will run the job for packages:
* `my-package` with version between 0.1.2 and 0.2
* `other-package` in version 1.1

Note OR (`|`) not yet supported.

### Reference implementations
* conda:
- MatchSpec (docstring): https://github.com/conda/conda/blob/aed799b56d9ee16a3653928527e2af6e41394e85/conda/models/match_spec.py#L73
- parsing: https://github.com/conda/conda/blob/aed799b56d9ee16a3653928527e2af6e41394e85/conda/models/match_spec.py#L569
- conda docs: https://github.com/conda/conda/blob/aed799b56d9ee16a3653928527e2af6e41394e85/docs/source/user-guide/concepts/pkg-specs.rst#package-match-specifications

* PEP440: https://www.python.org/dev/peps/pep-0440/#version-scheme

### CODE REVIEW

**Comment by wolfv** [comment]
Cool! I'll merge, but I'm afraid version comparison might not fully work. For example, when you have 3 versions like 

0.2.0
0.12.0
1.0.0

I'm not sure how string comparison is implemented in sql.

We might actually have to add the python function as extension to Postgres :)

**Comment by btel** [comment]
and how about defining the version as a composite type (major, minor, patch) and using this trick? :


https://docs-sqlalchemy.readthedocs.io/ko/latest/orm/composites.html#redefining-comparison-operations-for-composites

**Comment by wolfv** [comment]
yeah, i guess that would work, but also not for all cases. I think really the best & coolest would be to figure out how to properly integrate the spec into postgres :)

We could have a postgres extension, and a Python fallback.

File: plugins/quetz_transmutation/quetz_transmutation/api.py [hunk 1]
```diff
@@ -12,6 +12,8 @@
 from quetz.jobs.models import Job
 from quetz.pkgstores import PackageStore
 
+from .rest_models import PackageSpec
```
Comment (wolfv) [inline plugins/quetz_transmutation/quetz_transmutation/api.py:15]: @btel I think this file is missing?

**Comment by dhirschfeld** [comment]
This is a neat trick:
```sql
SELECT id, version
FROM   versions
ORDER  BY string_to_array(version, '.')::int[];
```
https://dba.stackexchange.com/a/74298

There's also an existing `semver` extension for postgres:
https://pgxn.org/dist/semver/doc/semver.html

*Disclaimer:* I haven't used it so YMMV! Also, PEP-440 is a bit different from SemVer IIUC

File: plugins/quetz_transmutation/quetz_transmutation/api.py (hunk 1 above)
Comment (btel) [reply plugins/quetz_transmutation/quetz_transmutation/api.py:15]: ah, sorry, I was tired. but why did the tests pass 🤔 ?!

**Comment by btel** [comment]
I would prefer to avoid using pg extensions. It complicates the installation and breaks compatibility with other databases. 

In this case we could come up with a neat sql query to do the trick.

**Pull request merged**